jsonschema/validators.py

```diff
@@ -259,17 +267,40 @@ def from_schema(cls, schema, *args, **kwargs):
 
         return cls(schema.get(u"id", u""), schema, *args, **kwargs)
 
+    def push_scope(self, scope):
+        self.scopes_stack.append(
+            self.urljoin_cache(self.resolution_scope, scope))
+
+    def pop_scope(self):
```

Comment:
Ech yep, right again. OK.

And yes to the error -- it's true it's unlikely, but if it's public API, someone else can use these, and do so incorrectly. Not a huge deal though.